dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapAdapter.java

```diff
@@ -207,6 +209,45 @@
      */
     private final AtomicReference<X509CertificateMapper> iCertificateMapperRef = new AtomicReference<X509CertificateMapper>();
 
+    private static final String KERB_SERVICE = "kerberosService";
+
+    private KerberosService kerberosService = null;
+
+    /**
+     * Set the KerberosService reference, Load the general Kerberos configuration so we can use the KerberosService and
+     * load the keytab/config attributes, if needed.
+     *
+     * @param ref
+     */
+    @Reference(name = KERB_SERVICE, service = KerberosService.class, policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+    protected void setKerberosService(KerberosService ref) {
+        kerberosService = ref;
+    }
+
+    /**
+     * Unset the KerberosService reference
+     *
+     * @param ref
+     */
+    protected void unsetKerberosService(KerberosService ref) {
+        if (kerberosService == ref) {
+            kerberosService = null;
```

Comment:
Because we added KerberosService as an ibmreference, it looks like as we do setkerberos then activate LdapAdapter and then we do the reverse on stop, deactive LdapAdapter then unsetKerberos. I will double check with Brent, but I think we're okay here.

I also checked what happens when you remove the Kerberos config altogether and it processes that as a modify and uses the defaultInstance of Kerberos config (which keeps the KerberosService available, but no keytab/config). That worked as expected -- we ran udpateKerberosService and redid the context pool.